### PR TITLE
How to buid a TODO app

### LINKED ISSUE #1723: How To Build a Todo app with flutter
Be sure to visit our [Resources Page](https://github.com/section-engineering-education/engineering-education/blob/master/new_contributors/resources-page.md) for tools, resources, and example articles to go over.

**NOTE: (Please ensure that you have only one open issue + linked pull request at a time. This will ensure that we complete the article in a timely manner from inception to publishing.)**

**NOTE: We tend to stray away or tend not to publish reviews/comparisons of commercial product offerings.**

### Proposed title of the article
Your title should be descriptive of the article/tutorial. Be Specific. *Use keyword research to gain your article's higher ranking.*

### Introduction paragraph(s):
Flutter is a tool kit created by Google. It is used to develop applications for android, IOS, Linux, Mac, Windows, Google fuchsia, and the web from a single codebase. This technology is changing the Software development industry by making application development faster and cheaper.
In this article, we will do an overview of what is possible with flutter by building a Todo app. The article will give the reader an understanding of how flutter works.
### Key takeaways: 
The reader will understand;
1. Flutter Widget
2. Stateful and stateless Widgets
3. Flutter and dart packages
### References: 
Please list links to any published content/research that you intend to use to support/guide this article.

### Templates to use as guides
- [How To Guide Template](https://github.com/section-engineering-education/engineering-education/blob/master/new_contributors/how-to-guide-template.md)
- [Software Review Template](https://github.com/section-engineering-education/engineering-education/blob/master/new_contributors/software-review-template.md)
- [Tutorial Template](https://github.com/section-engineering-education/engineering-education/blob/master/new_contributors/tutorial-template.md)

### PR DESCRIPTION
**Pull request by wobin1**
### Article checklist
**NOTE: (Please ensure that you have only one open issue + linked pull request at a time. This will ensure that we complete the article in a timely manner from inception to publishing.)**
#### Preliminary Checks - Formatting and Structure 
- [x] Does your article follow any of the suggested structure formats? - **see [example formats](https://github.com/section-io/engineering-education/blob/master/CONTRIBUTING.md#example-format-structure)**
- [x] Is your article properly formatted in Markdown? - **see [Github Markdown guide](https://github.com/adam-p/markdown-here/wiki/Markdown-Cheatsheet)**
- [x] Have you used the correct folder and file structure? - **see [Contribution Guidelines](https://github.com/section-io/engineering-education/blob/master/CONTRIBUTING.md)**
- [x] Is your article over 750 words? - **Tool**: [Wordcounter.net](https://wordcounter.net)
- [x] Have you used our preferred heading structure? - **[H3 (###) and up](https://github.com/adam-p/markdown-here/wiki/Markdown-Cheatsheet#headers)**
- [x] Does your article provide enough value and detail about your topic? - **Articles should be clear, accurate, and fully explained.**
- [x] Can your article be understood by beginners? - **Assume the audience is smart but has no prior exposure to the common terminology in your article.**
- [x] Have you included a hero image that is 600x400 pixels and have the copyright to use it? - **Tools:** [Pixlr Image Editor](https://pixlr.com/e) for resizing and [Unsplash](https://unsplash.com) for Creative Commons images
- [x] In your PR - add a comment that states **["This fixes #(enter your Idea Content Form #)"](https://youtu.be/TKJ4RdhyB5Y?t=278)** to link and close your Idea Content (issue) when the PR is merged.
      
#### Grammar & plagiarism checks
- [x] Have you spellchecked and grammar checked your article? - **Tools:** [Write&Improve](https://writeandimprove.com/?lang=en-GB) - [ProWritingAid](https://prowritingaid.com/) -  [Grammarly](https://grammarly.com) 
- [x] Please place your article through a 3rd party plagiarism checker? We suggest using Quetext, [this tool](https://www.quetext.com/) is free and has a daily limit. **We typically accept articles with 10% or less.**
- [x] Have you checked your article for readability? - **Tool:** [Hemmingway](http://www.hemingwayapp.com)
- [x] Have you added sources for quotes and images that aren't yours?

#### Technical checks
- [x] Are your code snippets properly formatted for syntax highlighting - **see [Syntax guide](https://gohugo.io/content-management/syntax-highlighting)**

### Contribution guidelines
For first-time contributors and for more details, **see [Contributing Guidelines](https://github.com/section-io/engineering-education/blob/master/CONTRIBUTING.md)**
This fixes #1723

### CODE REVIEW

**Comment by wobin1** [comment]
@hectorkambow  please if it is allowed, I will like @srishilesh to review my first PR.
I went through some of his reviews and Iwill like to learn under him. Thank you
Cc @srishilesh

**Comment by cyberShaw** [comment]
> @hectorkambow please if it is allowed, I will like @srishilesh to review my first PR.
> I went through some of his reviews and Iwill like to learn under him. Thank you
> Cc @srishilesh

And @srishilesh also has 5-6 articles assigned to him already.
Don't worry, ping me on Slack and I'll walk you through the PR

**Comment by srishilesh** [comment]
Hey @wobin1,

Thanks for asking.

As @WanjaMIKE said, I believe it's not allowed choose a reviewer.

@cyberShaw is a great peer reviewer, you can a learn a lot from him. In fact, I learnt from him.

**Comment by wobin1** [comment]
It's suppose to be in the file... But I still send it on slack. thank you
sir....

**Comment by cyberShaw** [comment]
@wobin1 Please maintain the image names in kebab-case without any spaces: https://www.theserverside.com/definition/Kebab-case

I've changed the image names in the commit: 449c68d

File: articles/how-to-build-a-todo-app/index.md [hunk 1]
```diff
@@ -0,0 +1,224 @@
...
+- [Flutter Widgets](#flutter-widgets)
+- [Stateless Widget](#stateless-widget)
+- [Statful Widget](#Stateful-widget)
+- [Conclusion](#conclusion)
+- [Futher reading](#futher-reading)
+
+
+### Introduction
+Over the years software developers have used frameworks to build apps for android, IOS, Desktop, and web, flutter is one of those frameworks.
+Flutter is a technology developed by Google to enable developers build apps for all platforms with a single code base. This technology is changing the Software development industry by making application development faster and cheaper.
```
Comment (cyberShaw) [inline articles/how-to-build-a-todo-app/index.md]:
```suggestion
Flutter is a technology developed by Google to enable developers to build apps for all platforms with a single code base. This technology is changing the Software development industry by making application development faster and cheaper.
```

File: articles/how-to-build-a-todo-app/index.md [hunk 2]
```diff
@@ -0,0 +1,224 @@
...
+In this article will use flutter to build a **To-do list** app.
```
Comment (cyberShaw) [inline articles/how-to-build-a-todo-app/index.md:16]:
```suggestion
In this article we will use flutter to build a **To-do list** app. 
```

**Review by cyberShaw — CHANGES_REQUESTED** [review]
I see that there are quite a lot of grammatical inconsistencies. 
Please run your article through a grammar/readability checker like [Grammarly](https://app.grammarly.com/) and fix those errors. And please pull changes before editing the article.

**Comment by cyberShaw** [comment]
I also noticed that all the images aren't loading and are corrupt. Could you please check on this? @wobin1

**Comment by wobin1** [comment]
@cyberShaw I have effected the changes you requested.

File: articles/how-to-build-a-todo-app/index.md [hunk 3]
```diff
@@ -0,0 +1,216 @@
...
+### Stateless Widget
+A stateless widget is a widget whose state cannot be changed once it is built. i.e. no amount of change in the variables, icons, buttons, or retrieving data can change the state of the app.
+A to-do app will always have to-do items added and remove, to achieve this we will need to implement a stateful widget.
+
+### Stateful Widget
+This type of widget is dynamic. This means it can change its appearance when it receives data or change appearance in response to events triggered.
+
+Edit your code to match the one in the picture below.
+
+![Stateful Widget](/engineering-education/how-to-build-a-todo-app/stateful.jpg)
```
Comment (cyberShaw) [inline articles/how-to-build-a-todo-app/index.md:85]: You've gotta add your code as code, and not as an image. Please delete this image and add the code within code blocks.

File: articles/how-to-build-a-todo-app/index.md [hunk 4]
```diff
@@ -0,0 +1,216 @@
...
+  
+You should notice that our stateful widgets class TodoList has two classes. This is to enable us to update our to-do list without losing our data. Now, let us add functionality to the state class. Add the below code to your state:
+
+![Code Picture](/engineering-education/how-to-build-a-todo-app/listText.jpg)
```
Comment (cyberShaw) [inline articles/how-to-build-a-todo-app/index.md:89]: Add the code here too

File: articles/how-to-build-a-todo-app/index.md [hunk 5]
```diff
@@ -0,0 +1,216 @@
...
+
+The alert dialog tells the user about situations that require confirmation. We are going to use the alert box to collect todo Items.
+To run the code we would have to update the build widget code to the one below.
+
+
+![Build Image](/engineering-education/how-to-build-a-todo-app/build.jpg)
+
+Your complete code should look the one below:
+
+```dart
```
Comment (cyberShaw) [inline articles/how-to-build-a-todo-app/index.md:116]: Please add code comments throughout your code and explain in detail how they work.

**Review by cyberShaw — CHANGES_REQUESTED** [review]
Please add the code snippets as text, and remove the code images. Pull changes before adding in your own. Please don't undo the changes I've made. 

**Please add in comments in the code and explain them in detail.**

We can move on to the final review after these changes have been made.

File: articles/how-to-build-a-todo-app/index.md (hunk 5 above)
Comment (wobin1) [reply articles/how-to-build-a-todo-app/index.md:276]: @cyberShaw I have effected the changes.

**Comment by wobin1** [comment]
I have affected the changes sir

**Comment by cyberShaw** [comment]
@wobin1 The indentation for many of the code snippets are messed up. Could you run them through a code formatter and update the snippets?

**Comment by cyberShaw** [comment]
I also noticed that the code comments were not concise enough, and were missing from a lot of the snippets. Please add them everywhere in the article

**Comment by wobin1** [comment]
code snippets updated.

**Comment by cyberShaw** [comment]
I'll do the review by today, @wobin1

**Comment by cyberShaw** [comment]
@wobin1 Please include a GitHub link in your article that contains the final code. Once you implement this, we can move on to final review.

**Comment by wobin1** [comment]
@cyberShaw i have added the link.

**Comment by cyberShaw** [comment]
Thank you for making the changes, @wobin1!
This article is ready for final review.

**Review by cyberShaw — APPROVED** [review]

**Comment by hectorkambow** [comment]
plagiarism check done

**Pull request merged**